websockets/send-or-receive-msg-websocket-client-endpoint.md

```diff
@@ -0,0 +1,45 @@
+---
+layout: default
+title: Send and receive messages to websocket client endpoint
+---
+
+### Problem
+
+You want to send and receive messages using the websocket client endpoint, which means, you need to send data with some static
+content, such as a string, to a websocket server and you, want to also receive the data from the webseocket server.
+
+### Solution
+
+Use the <code class="node">WEBSOCKET In</code> node to connect to a [public websocket server](https://www.websocket.org/echo.html){: new_window} and send the data to it , an <code class="node">Inject</code> node to include the static content, and a
+<code class="node">WEBSOCKET out</code> node to connect to the [public websocket server](https://www.websocket.org/echo.html){: new_window} and receive the data from it.
+
+#### Example
+
+![](/images/websockets/websockets-flow-001.png)
+
+{% raw %}
+~~~json
+[{"id":"2ee67bcf.e9ec74","type":"websocket out","z":"9330addb.1f80a8","name":"WS Echo Server","server":"","client":"ddf9ced0.198cd8","x":359.5,"y":142,"wires":[]},{"id":"4ad89394.6cb89c","type":"inject","z":"9330addb.1f80a8","name":"Say Hello","topic":"","payload":"Hello websockets","payloadType":"str","repeat":"","crontab":"","once":false,"x":150.5,"y":117,"wires":[["2ee67bcf.e9ec74"]]},{"id":"ddf9ced0.198cd8","type":"websocket-client","z":"","path":"ws://echo.websocket.org","wholemsg":"false"}]
+[{"id":"73d7301b.92abc","type":"debug","z":"9330addb.1f80a8","name":"debug","active":true,"console":"false","complete":"payload","x":324.5,"y":251,"wires":[]},{"id":"92491c57.ff3888","type":"websocket in","z":"9330addb.1f80a8","name":"WS Echo Server","server":"","client":"ddf9ced0.198cd8","x":150.5,"y":217,"wires":[["73d7301b.92abc"]]},{"id":"ddf9ced0.198cd8","type":"websocket-client","z":"","path":"ws://echo.websocket.org","wholemsg":"false"}]
```

Comment:
Currently this flow won't import for me... looks like this should be one array instead.
 
```
[{"id":"d81ec5df.bfa5b8","type":"websocket out","z":"3658255b.2041aa","name":"WS Echo Server","server":"","client":"8f146297.3f29","x":400,"y":760,"wires":[]},{"id":"4c3d1637.22f878","type":"inject","z":"3658255b.2041aa","name":"Say Hello","topic":"","payload":"Hello websockets","payloadType":"str","repeat":"","crontab":"","once":false,"x":191,"y":735,"wires":[["d81ec5df.bfa5b8"]]},{"id":"2308f96f.33b806","type":"debug","z":"3658255b.2041aa","name":"debug","active":true,"console":"false","complete":"payload","x":365,"y":869,"wires":[]},{"id":"faa3cf2b.318bf","type":"websocket in","z":"3658255b.2041aa","name":"WS Echo Server","server":"","client":"8f146297.3f29","x":191,"y":835,"wires":[["2308f96f.33b806"]]},{"id":"8f146297.3f29","type":"websocket-client","z":"","path":"ws://echo.websocket.org","wholemsg":"false"}]
```